SignalServiceKit/src/Storage/OWSStorage.h

```diff
@@ -0,0 +1,53 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import <YapDatabase/YapDatabaseConnection.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol OWSDatabaseConnectionDelegate <NSObject>
+
+- (BOOL)isDatabaseInitialized;
+
+@end
+
+#pragma mark -
+
+@class YapDatabaseExtension;
+
+@interface OWSStorage : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+- (instancetype)initStorage NS_DESIGNATED_INITIALIZER;
+
+- (void)setDatabaseInitialized;
+
++ (void)resetAllStorage;
+
+// TODO: Deprecate?
+- (nullable YapDatabaseConnection *)newDatabaseConnection;
+
+// TODO: Deprecate.
+@property (nullable, nonatomic, readonly) YapDatabaseConnection *dbReadConnection;
+@property (nullable, nonatomic, readonly) YapDatabaseConnection *dbReadWriteConnection;
```

Comment:
Hopefully these connections can soon be moved to `TSStorageManager` or removed completely, but I'll try to make progress towards the SAE first.